docs/UA-service-API.md

```diff
@@ -0,0 +1,437 @@
+# User Agent Service REST API
+
+This document provides protocol-level details of the User Agent
+Service REST API.
+
+# Overview
+
+The User Agent Service provides two sub-components intended to back an
+abstract conception of browsing: an HTTP REST API to allow bulk access
+to data, and a light WebSocket API to push incremental changes to
+clients.  The WebSocket API will be documented separately.
+
+The abstract conception of browsing introduces two high-level
+concepts, _scopes_ and _sessions_.  Roughly speaking, a _scope_
+identifies a logical or conceptual grouping of browsing behaviour.
+For Tofino, a scope corresponds to an open browser window; for a
+mobile search activity, a scope might correspond to a single
+interaction with the main activity.  A _session_ identifies a single
+browsing context.  It takes place within a scope (put differently,
+scopes encapsulate sessions) and may have a logical _ancestor
+session_.  For Tofino, a session corresponds to a single tab; a tab
+spawned by clicking a link with "target=_blank" would correspond to a
+new session with the original tab as it's ancestor session.
+
+Scopes are currently allocated as new sessions in Tofino.
```

Comment:
"Scopes are currently allocated as new sessions in Tofino" - not sure what this means.  We treat a scope the same as a session?